pkg/networkmanager/firewall.jsx

```diff
@@ -0,0 +1,167 @@
+
+    var tcp = props.service.ports.filter(p => p.protocol.toUpperCase() == 'TCP');
+    var udp = props.service.ports.filter(p => p.protocol.toUpperCase() == 'UDP');
+
+    function onRemoveService(event) {
+        if (event.button !== 0)
+            return;
+
+        props.onRemoveService(props.service.id);
+        event.stopPropagation();
+    }
+
+    var columns = [
+        { name: props.service.name, header: true },
+        <div>
+            { tcp.map(p => p.port).join(', ') }
+        </div>,
+        <div>
+            { udp.map(p => p.port).join(', ') }
+        </div>,
+        <button className="btn btn-danger pficon pficon-delete"
+                onClick={onRemoveService}/>
+    ];
+
+    var tabs = [
+        { name: _("Details"), renderer: () => <p>{props.service.description}</p> }
+    ];
+
+    return <ListingRow key={props.service.id}
+                       columns={columns}
+                       tabRenderers={tabs}/>;
+}
+
+export class Firewall extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            installed: firewall.installed,
+            enabled: firewall.enabled,
+            services: firewall.services,
+            enabledServices: firewall.enabledServices
+        };
+
+        this.onFirewallChanged = this.onFirewallChanged.bind(this);
+        this.onRemoveService = this.onRemoveService.bind(this);
+    }
+
+    onFirewallChanged() {
+        this.setState({
+            installed: firewall.installed,
+            enabled: firewall.enabled,
+            services: firewall.services,
+            enabledServices: firewall.enabledServices
+        });
+    }
+
+    onSwitchChanged(value) {
+        if (value)
+            firewall.enable();
+        else
+            firewall.disable();
+    }
+
+    onRemoveService(service) {
+        firewall.removeService(service);
+    }
+
+    componentDidMount() {
+        firewall.addEventListener("changed", this.onFirewallChanged);
+    }
+
+    componentWillUnmount() {
+        firewall.removeEventListener("changed", this.onFirewallChanged);
+    }
+
+    render() {
+        function go_up() {
+            if (!event || event.button !== 0)
+                return;
+
+            cockpit.jump("/network", cockpit.transport.host);
+        }
+
+        if (!this.state.installed) {
+            return (
+                <EmptyState title={_("Firewall is not available")} icon="fa fa-exclamation-circle">
+                    <p translatable="yes">Please install the <tt>firewalld</tt> package.</p>
+                </EmptyState>
+            );
+        }
+
+        var services = [...this.state.enabledServices].map(id => this.state.services[id]);
+        services.sort((a, b) => a.name - b.name);
+
+        return (
+            <div className="container-fluid page-ct">
+                <ol className="breadcrumb">
+                    <li><a onClick={go_up}>{_("Network")}</a></li>
```

Comment:
Just a flyby comment that this is called "Networking" in the menu and on the breadcrumb on the other sub-pages.

Reply:
Indeed. Thanks for the catch! It'll be fixed with the next push.